sync-core/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -74,6 +77,15 @@
 
     public final HashMap<String, String> requestProperties;
 
+    public final LinkedList<HttpConnectionRequestFilter> requestFilters;
+    public final LinkedList<HttpConnectionResponseFilter> responseFilters;
+
+    //we need to store the response code separately so we don't send a request
+    //before we are ready
+    private int responseCode = -1;
```

Comment:
Sadly there isn't calling for the response code from the URL connection object will cause a request to get sent (if one hasn't been made already), breaking everything else.